include/seqan3/io/sequence/sequence_file_out.hpp

```diff
@@ -0,0 +1,900 @@
+     *
+     * data_storage_t data_storage; // a global or globally used variable in your program
+     *
+     * // ... in your file writing function:
+     *
+     * sequence_file_out fout{"/tmp/my.fasta"};
+     *
+     * fout = std::tie(data_storage.sequences, data_storage.ids);
+     * ```
+     */
+    template <typename typelist, typename field_ids>
+    sequence_file_out & operator=(record<typelist, field_ids> const & r)
+    {
+        write_columns(detail::range_wrap_ignore(detail::get_or_ignore<field::SEQ>(r)),
+                      detail::range_wrap_ignore(detail::get_or_ignore<field::ID>(r)),
+                      detail::range_wrap_ignore(detail::get_or_ignore<field::QUAL>(r)),
+                      detail::range_wrap_ignore(detail::get_or_ignore<field::SEQ_QUAL>(r)));
+    }
+
+    /*!\brief            Write columns (wrapped in a std::tuple) to the file.
+     * \tparam arg_types The column types, each type must be a range-of-range.
+     * \param[in] t      The tuple of columns.
+     *
+     * \details
+     *
+     * The columns are assumed to correspond to the field IDs given in selected_field_ids, however passing less
+     * is accepted if the format does not require all of them.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of the columns.
+     *
+     * ### Exceptions
+     *
+     * Basic exception safety.
+     *
+     * ### Example
+     *
+     * ```cpp
+     *
+     * struct data_storage_t
+     * {
+     *     concatenated_sequences<dna5_vector>  sequences;
+     *     concatenated_sequences<std::string>  ids;
+     * };
+     *
+     * data_storage_t data_storage; // a global or globally used variable in your program
+     *
+     * // ... in your file writing function:
+     *
+     * sequence_file_out fout{"/tmp/my.fasta"};
+     *
+     * fout = std::tie(data_storage.sequences, data_storage.ids);
+     * ```
+     */
+    template <typename ... arg_types>
+    sequence_file_out & operator=(std::tuple<arg_types...> const & t)
+    {
+        // index_of might return npos, but this will be handled well by get_or_ignore (and just return ignore)
+        write_columns(
+            detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::SEQ)>(t)),
+            detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::ID)>(t)),
+            detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::QUAL)>(t)),
+            detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::SEQ_QUAL)>(t)));
+    }
+    //!\}
+
+    //!\brief The options are public and its members can be set directly.
+    sequence_file_out_options options;
+
+    /*!\cond DEV
+     * \brief Expose a reference to the underlying stream object. [public, but not documented as part of the API]
+     */
+    stream_type & get_stream()
+    {
+        return stream;
+    }
+    //!\endcond
+protected:
+    //!\privatesection
```

Comment:
I will never understand when to use this and when not. Is protected: not already marking it as not public. Should be documented as private also it is protected?

Reply:
protected is protectedsection by default :laughing: 

no, seriously, the issue is that protected stuff is shown in our user-documentation, but in most cases our protected stuff is DEV-only (you might need it when doing library side inheritance, but you don't actually want users to inherit from this class).

Reply:
ok, thanks for enlighten me